modules/plottinggl/src/processors/parallelcoordinates/parallelcoordinates.cpp

```diff
@@ -470,45 +519,41 @@ void ParallelCoordinates::buildLineMesh(
     recreateLines_ = false;
 }
 
-void ParallelCoordinates::drawAxis(
-    size2_t size, const std::vector<ParallelCoordinatesAxisSettingsProperty *> &enabledAxis) {
-    axisShader_.activate();
-
-    axisShader_.setUniform("dims", ivec2(size));
-    axisShader_.setUniform("spacing", margins_.getAsVec4());
-    axisShader_.setUniform("color", axisColor_.get());
-    axisShader_.setUniform("hoverColor", axisHoverColor_.get());
-    axisShader_.setUniform("selectedColor", axisSelectedColor_.get());
+void ParallelCoordinates::drawAxis(size2_t size, const std::vector<ColumnAxis *> &enabledAxis) {
+    const size2_t lowerLeft(margins_.getLeft(), margins_.getBottom());
+    const size2_t upperRight(size.x - 1 - margins_.getRight(), size.y - 1 - margins_.getTop());
 
+    const auto padding = 0;
     float dx = 1.0f / (enabledAxis.size() - 1);
 
-    size_t axisCounter = 0;
-    size_t activeAxisCounter = 0;
-    for (auto &p : axisVector_) {
-        if (p->isChecked()) {
-            float x = activeAxisCounter * dx;
-            axisShader_.setUniform("x", x);
-            axisShader_.setUniform("hover", 0);
-            axisShader_.setUniform("selected", 0);
-            if (hoveredAxis_ == axisCounter)
-                axisShader_.setUniform("hover", 1);
-            if (brushingAndLinking_.isColumnSelected(axisCounter))
-                axisShader_.setUniform("selected", 1);
-
-            axisShader_.setUniform("pickColor", axisPicking_.getColor(axisCounter));
-            axisDrawer_->draw();
-            activeAxisCounter++;
+    for (size_t i = 0; i < enabledAxis.size(); i++) {
```

Comment:
this way of setting properties from properties is huge code smell, state should __not__ be duplicated.